src/mito2/src/worker/handle_open.rs

```diff
@@ -66,6 +66,10 @@ impl<S: LogStore> RegionWorkerLoop<S> {
         request: RegionOpenRequest,
         sender: OptionOutputTx,
     ) {
+        if self.opening_regions.is_region_exists(region_id) {
+            sender.send(Ok(0));
+            return;
+        };
```

Comment:
Maybe we should not return ok directly. If the opener fails in the background, then the region isn't opened but we return ok.

One solution is adding the sender to the `opening_regions` map and notifying the result later.